overlord/devicestate/devicestate_install_mode_test.go

```diff
@@ -1148,6 +1160,9 @@ func (s *deviceMgrInstallModeSuite) TestInstallWithInstallDeviceHookExpTasks(c *
 
 	c.Assert(hooksCalled, HasLen, 1)
 	c.Assert(hooksCalled[0].HookName(), Equals, "install-device")
+
+	// ensure systemctl daemon-reload gets called at least once
+	c.Assert(s.SystemctlDaemonReloadCalls, testutil.IntGreaterEqual, 1)
```

Comment:
(nitpick) under what circumstances would it be greater than once? i.e. why not just check for exactness here (and below)?

Reply:
Glad you commented on this because it was something I was hoping for feedback on. 

I don't think that at the moment it gets called more than once. My motivation here was that I wanted to make the test less brittle. I don't *actually* care if it gets called more than once, I just want to make sure it gets called under certain circumstances. So in the future, if some other code change changes the behavior (and calls daemon-reload a second time, perhaps at some other point during the handler, maybe before the hook runs?) in a way that doesn't break the "spirit" of the test, the test still passes. This is also why I didn't add any negative checks (i.e. `SystemctlDaemonReloadCalls, Equal, 0`) in the other cases.

Otoh, if there was another behavioral change (for instance, if a call to `systemctl daemon-reload` gets added *before* the install-device hook actually executes, and this later call gets broken somehow) this could mask it.

So I think there are pros and cons either way, and am interested in which you think might be better.

Reply:
I think it's ok to check for 1 for now. It's a bit unlikely that task will start reloading more, if it does we can reconsider once we get there

Reply:
I updated the test now to use Equal as it's more in line what we do elsewhere.